tests/testthat/test-serializer.R

```diff
@@ -268,3 +268,75 @@ test_that("Windows encoding", {
 
   expect_type(ser$deserialize(), "list") # could fail on Windows
 })
+
+# Compressed -------------------------------------------------------------------
+
+test_that("generates correct path", {
+  aa <- Compressed$new("path", "name")
+  expect_equal(aa$path, "path/name.qs2")
+})
+
+test_that("generates expected compressed data", {
+  local_vcr_configure(json_pretty = TRUE)
+  local_mocked_bindings(
+    cur_time = function(tz) "2024-01-01 12:00:00",
+    pkg_versions = function() "<package_versions>"
+  )
+
+  request <- Request$new(method = "GET", uri = "http://example.com")
+  response <- VcrResponse$new(status = 200L, list(name = "val"), body = "body")
+  interaction <- list(request = request, response = response)
+
+  ser <- Compressed$new(withr::local_tempdir(), "serialize")
+  ser$serialize(list(interaction))
+
+  expect_snapshot(writeLines(qs2::qs_read(ser$path)))
+})
+
+test_that("Compressed usage", {
+  skip_on_cran()
+  local_vcr_configure(
+    dir = withr::local_tempdir(),
+    serialize_with = "compressed"
+  )
+
+  # does one request work?
```

Comment:
I think it would be good to think through your testing goals here, as I think this code mostly tests mechanics outside of serialization. I think a better test (which we might also want to port to YAML and JSON) would be to test that deserialisation + serialisation is idempotent.

Reply:
I'll have a think about that and try replacing this

Reply:
@hadley thoughts on the tests now?